### PR TITLE
feat: added Rename button to File Explorer

### PR DESCRIPTION
**Pull request by Toxer10**
This PR...

## Changes

- added Rename file or folder functionality to File Explorer
- added Select All hotkey (ctrl+a or cmd+a)

## Screenshots

image

### CODE REVIEW

**Comment by erdkse** [comment]
lgtm! And liked this!!

Screen Shot 2021-11-19 at 10 43 49

**Review by erdkse — APPROVED** [review]

File: src/components/organisms/FileTreePane/FileTreePane.tsx [hunk 1]
```diff
@@ -289,7 +288,7 @@ interface TreeItemProps {
   setProcessingEntity: Dispatch<SetStateAction<ProcessingEntity>>;
   processingEntity: ProcessingEntity;
   onDeleting: (args: DeleteEntityCallback) => void;
-  onRenaming: (absolutePath: string) => void;
+  onRenaming: (absolutePath: string, os: any) => void;
```
Comment (devcatalin) [inline src/components/organisms/FileTreePane/FileTreePane.tsx:291]: should this be `os: string` ?

File: src/components/organisms/FileTreePane/FileTreePane.tsx [hunk 2]
```diff
@@ -566,8 +570,8 @@ const FileTreePane = () => {
     }, 2000);
   };
 
-  const onRenaming = (absolutePathToEntity: string) => {
-    dispatch(openRenameEntityModal(absolutePathToEntity));
+  const onRenaming = (absolutePathToEntity: string, os: string) => {
```
Comment (devcatalin) [inline src/components/organisms/FileTreePane/FileTreePane.tsx:573]: could you rename `onRenaming` to `onRename` and do the same for `onDeleting` to `onDelete` ?

File: src/models/appconfig.ts [hunk 3]
```diff
@@ -56,6 +56,7 @@ interface AppConfig {
     data: any;
   };
   kubeConfig: KubeConfig;
+  os: NodeJS.Platform;
```
Comment (devcatalin) [inline src/models/appconfig.ts:59]: let's rename this to `osPlatform`

**Comment by devcatalin** [comment]
Looks good! Let's do one last change, instead of creating two redux actions: `setScanExcludesUpdated` and `setScanExcludesOutdated`, you could create a single `setScanExcludesStatus` action.
You could rename `isScanExcludesUpdated` to `scanExcludesStatus` and set the type to `'applied' | 'outdated'` , then use the `setScanExcludesStatus` to toggle between `applied` or `outdated`. Thanks!

**Comment by Toxer10** [comment]
> Looks good! Let's do one last change, instead of creating two redux actions: `setScanExcludesUpdated` and `setScanExcludesOutdated`, you could create a single `setScanExcludesStatus` action. You could rename `isScanExcludesUpdated` to `scanExcludesStatus` and set the type to `'applied' | 'outdated'` , then use the `setScanExcludesStatus` to toggle between `applied` or `outdated`. Thanks!

Done.

File: src/models/appconfig.ts (hunk 3 above)
Comment (Toxer10) [reply src/models/appconfig.ts]: Done.

File: src/components/organisms/FileTreePane/FileTreePane.tsx (hunk 2 above)
Comment (Toxer10) [reply src/components/organisms/FileTreePane/FileTreePane.tsx]: Done.

File: src/components/organisms/FileTreePane/FileTreePane.tsx (hunk 1 above)
Comment (Toxer10) [reply src/components/organisms/FileTreePane/FileTreePane.tsx]: Done.

**Pull request merged**